src/libs/ajax/User.ts

```diff
@@ -0,0 +1,398 @@
+import _ from 'lodash/fp';
+import * as qs from 'qs';
+import {
+  authOpts,
+  fetchBond,
+  fetchEcm,
+  fetchOk,
+  fetchOrchestration,
+  fetchRex,
+  fetchSam,
+  jsonBody,
+} from 'src/libs/ajax/ajax-common';
+import { getConfig } from 'src/libs/config';
+import { getTerraUser } from 'src/libs/state';
+import * as Utils from 'src/libs/utils';
+
+export interface SamUserRegistrationStatusResponse {
+  userSubjectId: string;
+  userEmail: string;
+  enabled: boolean;
+}
+
+export interface SamUserTosStatusResponse {
+  userInfo: {
+    userSubjectId: string;
+    userEmail: string;
+  };
+  enabled: {
+    ldap: boolean;
+    allUsersGroup: boolean;
+    google: boolean;
+  };
+}
+
+export interface SamUserTosComplianceStatusResponse {
+  userId: string;
+  userHasAcceptedLatestTos: boolean;
+  permitsSystemUsage: boolean;
+}
+
+export interface OrchestrationUserProfileResponse {
+  userId: string;
+  keyValuePairs: { key: any; value: any }[];
```

Comment:
`unknown` might be preferable over `any` here. `any` effectively turns off type checking, while `unknown` requires code to determine what type a value is before using it.

```suggestion
  keyValuePairs: { key: unknown; value: unknown }[];
```

https://www.typescriptlang.org/docs/handbook/2/functions.html#unknown

Also, can the keys really contain any type? I would have guessed they would always be strings.

Reply:
My mistake, I had looked at this before (https://firecloud-orchestration.dsde-dev.broadinstitute.org/#/Profile/getAll ) and I thought I was running into build errors. Turns out it was fine. 

Both the key and value are `strings`